### PR TITLE
[tensorflow, pytorch] [build] [test] [ec2, ecs, eks, sagemaker] Add EFA stack and tests

### PR DESCRIPTION
**Pull request by jeet4320**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jeet4320** [comment]
dlc-pr-tensorflow-2_dlc-pr-sagemaker-test 
 ```
import smddpcommon as hc 
ImportError: libboost_thread.so.1.73.0: cannot open shared object file: No such file or directory
```

File: pytorch/training/docker/1.8/py3/cu111/Dockerfile.gpu [hunk 1]
```diff
@@ -220,6 +222,22 @@ RUN mv $OPEN_MPI_PATH/bin/mpirun $OPEN_MPI_PATH/bin/mpirun.real \
...
+  && cd /tmp/efa-ofi-nccl \
+  && git clone https://github.com/aws/aws-ofi-nccl.git -b v${BRANCH_OFI} \
+  && cd aws-ofi-nccl \
+  && ./autogen.sh \
+  && ./configure --with-libfabric=/opt/amazon/efa \
+  --with-mpi=/opt/amazon/openmpi \
+  --with-cuda=/usr/local/cuda \
+  --with-nccl=/usr/local --prefix=/usr/local \
+  && make \
+  NVCC_GENCODE="-gencode=arch=compute_70,code=sm_70 -gencode=arch=compute_75,code=sm_75 -gencode=arch=compute_80,code=sm_80 -gencode=arch=compute_80,code=compute_80" \
```
Comment (saimidu) [inline pytorch/training/docker/1.8/py3/cu111/Dockerfile.gpu:237]: The compute types listed here do not exactly overlap with the versions listed at line 29:
```
ENV TORCH_CUDA_ARCH_LIST="3.7 5.0 7.0+PTX 8.0"
```
Does this have any implications for NCCL support on any instance types, and does this need sm_75 support to be added to the PyTorch framework binary?

File: tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu [hunk 2]
```diff
@@ -238,6 +244,22 @@ RUN  wget -nv https://github.com/rapidsai/rmm/archive/v${RMM_VERSION}.tar.gz \
...
+  && cd /tmp/efa-ofi-nccl \
+  && git clone https://github.com/aws/aws-ofi-nccl.git -b v$BRANCH_OFI \
+  && cd aws-ofi-nccl \
+  && ./autogen.sh \
+  && ./configure --with-libfabric=/opt/amazon/efa \
+  --with-mpi=/opt/amazon/openmpi \
+  --with-cuda=/usr/local/cuda \
+  --with-nccl=/usr/local --prefix=/usr/local \
+  && make \
+  NVCC_GENCODE="-gencode=arch=compute_70,code=sm_70 -gencode=arch=compute_75,code=sm_75 -gencode=arch=compute_80,code=sm_80 -gencode=arch=compute_80,code=compute_80" \
```
Comment (saimidu) [inline tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu:259]: Please confirm that these GPU archs are a subset of those supported by TF framework on this dockerfile.

File: test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py [hunk 3]
```diff
@@ -28,6 +28,7 @@
 PT_EC2_MULTI_GPU_INSTANCE_TYPE = get_ec2_instance_type(
     default="g3.8xlarge", processor="gpu", filter_function=ec2_utils.filter_only_multi_gpu,
 )
+PT_EC2_GPU_EFA_INSTANCE_TYPE = get_ec2_instance_type(default="p3dn.24xlarge", processor="gpu", efa=True)
```
Comment (saimidu) [inline test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py:31]: If EFA tests are meant to be run only on `*dn.*` instance types, and nothing else, then these tests need to be skipped accordingly.

While this test is unlikely to run on a job that is configured to use an instance type that doesn't support it, we need to account for the edge cases where p3dn.24xlarge is not the instance type being used, and the `not efa` flag is not passed as input to the `pytest` command.

Therefore, I recommend inserting a `filter_function` like `ec2_utils.filter_only_efa_instances` to this parametrization of the `ec2_instance_type`. This filter function doesn't currently exist, but you can easily add this.

```suggestion
PT_EC2_GPU_EFA_INSTANCE_TYPE = get_ec2_instance_type(default="p3dn.24xlarge", processor="gpu", filter_function=ec2_utils.filter_only_efa_instances, efa=True)
```

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py [hunk 4]
```diff
@@ -26,29 +27,25 @@
...
-
-def can_run_smdataparallel(ecr_image):
+def can_run_smmodelparallel(ecr_image):
     _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
     image_cuda_version = get_cuda_version_from_tag(ecr_image)
     return Version(image_framework_version) in SpecifierSet(">=1.6") and Version(image_cuda_version.strip("cu")) >= Version("110")
 
-
-def can_run_smmodelparallel(ecr_image):
+def can_run_smmodelparallel_efa(ecr_image):
```
Comment (saimidu) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py:45]: Not a part of review: I wonder if this could be a `LABEL` on the image that gets built.

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+@pytest.mark.multinode(2)
+@pytest.mark.integration("smdataparallel")
+@pytest.mark.parametrize('instance_types', ["ml.p4d.24xlarge"])
+@pytest.mark.skip_cpu
+@pytest.mark.efa()
+def test_smdataparallel_throughput(n_virginia_sagemaker_session, framework_version, n_virginia_ecr_image, instance_types, tmpdir):
+    with timeout(minutes=DEFAULT_TIMEOUT):
+        image_cuda_version = get_cuda_version_from_tag(n_virginia_ecr_image)
+        if Version(framework_version) < Version("1.8.1") or image_cuda_version != "cu111":
+            pytest.skip("EFA is only supported on CUDA 11, and on PyTorch 1.8.1 or higher")
```
Comment (saimidu) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py:50]: Please make this a `validate_or_skip_*(ecr_image)` function.

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+@pytest.mark.efa()
+@pytest.mark.parametrize('instance_types', ["ml.p3.16xlarge", "ml.p4d.24xlarge"])
+def test_smdataparallel_mnist(n_virginia_sagemaker_session, framework_version, n_virginia_ecr_image, instance_types, tmpdir):
+    """
+    Tests smddprun command via Estimator API distribution parameter
+    """
+    with timeout(minutes=DEFAULT_TIMEOUT):
+        image_cuda_version = get_cuda_version_from_tag(n_virginia_ecr_image)
+        if Version(framework_version) < Version("1.8.0") or image_cuda_version != "cu111":
+            pytest.skip("EFA is only supported on CUDA 11, and on PyTorch 1.8.1 or higher")
```
Comment (saimidu) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py:116]: Please make this a `validate_or_skip_*(ecr_image)` function.

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_mnist.py [hunk 7]
```diff
@@ -191,6 +191,7 @@ def test_smdebug(sagemaker_session, ecr_image, instance_type, framework_version)
 @pytest.mark.model("mnist")
 @pytest.mark.skip_cpu
 @pytest.mark.skip_py2_containers
+@pytest.mark.skip(reason="Consider reworking these tests after re:Invent releases are done")
```
Comment (saimidu) [inline test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_mnist.py:194]: Please remove this.

File: test/test_utils/sagemaker.py [hunk 8]
```diff
@@ -121,6 +121,7 @@ def generate_sagemaker_pytest_cmd(image, sagemaker_test_type):
     else:
         integration_path = os.path.join("integration", sagemaker_test_type)
 
+
```
Comment (saimidu) [inline test/test_utils/sagemaker.py:124]: nit: Can remove extra line space here.

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_mnist.py (hunk 7 above)
Comment (jeet4320) [reply test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_mnist.py:194]: this is outdated test and will be fixed afterwards. https://github.com/aws/deep-learning-containers/pull/1044/files#diff-de6e76f81c56a48b429095acaa583f8bf9f9395ed7efd74b7cf24df89951f6eeL198

File: pytorch/training/docker/1.8/py3/cu111/Dockerfile.gpu [hunk 9]
```diff
@@ -30,8 +29,10 @@ ENV PATH /opt/conda/bin:$PATH
 ENV TORCH_CUDA_ARCH_LIST="3.7 5.0 7.0+PTX 8.0"
 ENV TORCH_NVCC_FLAGS="-Xfatbin -compress-all"
 ENV CUDNN_VERSION=8.0.5.39
-ENV NCCL_VERSION=2.8.4
+ENV NCCL_VERSION=2.7.8
```
[the remark this answers is not in the document]
Comment (saimidu) [reply pytorch/training/docker/1.8/py3/cu111/Dockerfile.gpu:32]: Received confirmation. Resolving conversation.

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py (hunk 4 above)
Comment (jeet4320) [reply test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py:45]: didnt get this

File: test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py (hunk 3 above)
Comment (jeet4320) [reply test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py:31]: we dont need this code, so removed it as I had added sample efa test but we didnt follow that approach and defined fixtures .

File: tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu (hunk 2 above)
Comment (jeet4320) [reply tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu:259]: removed this, we added this to save build time to build image just for those archs, but I think we dont need this code as it is difficult to maintain

File: pytorch/training/docker/1.8/py3/cu111/Dockerfile.gpu (hunk 1 above)
Comment (jeet4320) [reply pytorch/training/docker/1.8/py3/cu111/Dockerfile.gpu:237]: same as below

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_mnist.py (hunk 7 above)
Comment (jeet4320) [reply test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_mnist.py:194]: we want to keep it and skip it for now.

File: tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu (hunk 2 above)
Comment (saimidu) [reply tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu]: We need to minimize DLC image sizes as well, and if constraining the list of supported archs does this for us, then we should prefer to constrain the list in some way or the other.

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py (hunk 4 above)
Comment (saimidu) [reply test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py:45]: Please ignore.

**Comment by jeet4320** [comment]
All ECS, EKS, EC2 tests are green, so will skip it for next commit as it wont be related to Dockerfile builds.
Also all Mxnet build and tests are green.

File: tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu [hunk 10]
```diff
@@ -115,10 +115,11 @@ RUN wget https://sourceforge.net/projects/boost/files/boost/1.73.0/boost_1_73_0.
  && tar -xzf boost_1_73_0.tar.gz \
  && cd boost_1_73_0 \
  && ./bootstrap.sh \
- && ./b2 threading=multi --prefix=/usr -j 64 cxxflags=-fPIC cflags=-fPIC install \
+ && ./b2 threading=multi --prefix=/usr -j 64 cxxflags=-fPIC cflags=-fPIC install || true \
  && cd .. \
  && rm -rf boost_1_73_0.tar.gz \
- && rm -rf boost_1_73_0
+ && rm -rf boost_1_73_0 \
+ && cd /usr/include/boost
```
Comment (saimidu) [inline tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu:122]: Is this a validation step to see if the build succeeded?

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py [hunk 11]
```diff
@@ -35,11 +35,14 @@ def can_run_smdataparallel(ecr_image):
     return Version(image_framework_version) in SpecifierSet(">=1.6") and Version(
         image_cuda_version.strip("cu")) >= Version("110")
 
+
 def validate_or_skip_smdataparallel_efa(ecr_image):
-    image_cuda_version = get_cuda_version_from_tag(n_virginia_ecr_image)
-    if Version(framework_version) < Version("1.8.1") or image_cuda_version != "cu111":
+    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+    image_cuda_version = get_cuda_version_from_tag(ecr_image)
+    if Version(image_framework_version) < Version("1.8.1") or image_cuda_version != "cu111":
```
Comment (saimidu) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py:42]: Please change the `image_cuda_version` comparison to be similar to the type of code in lines 35-36 of this script.

File: test/testrunner.py [hunk 12]
```diff
@@ -47,8 +47,8 @@ def run_sagemaker_local_tests(images):
     sm_tests_tar_name = "sagemaker_tests.tar.gz"
     run(f"tar -cz --exclude='*.pytest_cache' --exclude='__pycache__' -f {sm_tests_tar_name} {sm_tests_path}")
     ec2_client = boto3.client("ec2", config=Config(retries={"max_attempts": 10}), region_name=DEFAULT_REGION)
-    for image in images:
-        sm_utils.execute_local_tests(image, ec2_client)
+    # for image in images:
+    #     sm_utils.execute_local_tests(image, ec2_client)
```
Comment (saimidu) [inline test/testrunner.py:51]: Why is this commented out? Also, this is not the latest version of the sm local code. Please merge from master to get the latest version of this code.

File: tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu (hunk 10 above)
Comment (jeet4320) [reply tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu:122]: to see if boost installation succeeded

File: test/testrunner.py (hunk 12 above)
Comment (jeet4320) [reply test/testrunner.py:51]: It is already green on PR, so reran just SM remote tests.

Comment (jeet4320) [reply test/testrunner.py:51]: This PR was created base on latest master available at that time, and master is going to change most probably. If we keep same rerunnning tests again for no Dockerfile change, PR will get delayed

**Comment by jeet4320** [comment]
@saimidu As all tests are green except two skipped one, I'm going to run just those tests on this PR by add code which will be reverted.

**Comment by jeet4320** [comment]
All SM, EKS, EC2, ECS tests are green now. 

For last commit, some are failing which are failing due to capacity issue or need retry due to flakiness

File: tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu [hunk 13]
```diff
@@ -93,31 +96,35 @@ RUN apt-get update && apt-get install -y --no-install-recommends --allow-unauthe
...
+  && rm -rf /tmp/nccl
+
+# Install EFA along with AWS OPEN_MPI
+RUN mkdir /tmp/efa \
+  && cd /tmp/efa \
+  && curl -O https://s3-us-west-2.amazonaws.com/aws-efa-installer/aws-efa-installer-$EFA_VERSION.tar.gz \
+  && tar -xf aws-efa-installer-$EFA_VERSION.tar.gz \
+  && cd aws-efa-installer \
+  && ./efa_installer.sh -y --skip-kmod -g \
+  && rm -rf /tmp/efa
```
Comment (yselivonchyk) [inline tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu:112]:
```suggestion
  && rm -rf /tmp/efa
  && rm -rf /tmp/aws-efa-installer-$EFA_VERSION.tar.gz
```

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smdataparallel.py [hunk 14]
```diff
@@ -27,23 +28,44 @@
...
+def validate_or_skip_smdataparallel(ecr_image):
+    if not can_run_smdataparallel(ecr_image):
+        pytest.skip("Data Parallelism is supported on CUDA 11 on TensorFlow 2.3.1 and above")
+
+
+def can_run_smdataparallel(ecr_image):
+    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+    image_cuda_version = get_cuda_version_from_tag(ecr_image)
+    return Version(image_framework_version) in SpecifierSet(">=2.3.1") and Version(
+        image_cuda_version.strip("cu")) >= Version("110")
```
Comment (saimidu) [inline test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smdataparallel.py:41]: nit: You can just compare two versions with `Version(x) >= Version(y)`, but using `SpecifierSet`s is also good.

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smdataparallel.py [hunk 15]
```diff
@@ -70,10 +93,8 @@ def test_smdataparallel_mnist(instance_types, n_virginia_ecr_image, py_version,
     """
     Tests smddprun command via Estimator API distribution parameter
     """
-    _, image_framework_version = get_framework_and_version_from_tag(n_virginia_ecr_image)
-    image_cuda_version = get_cuda_version_from_tag(n_virginia_ecr_image)
-    if Version(image_framework_version) < Version("2.3.1") or image_cuda_version != "cu110":
-        pytest.skip("Data Parallelism is only supported on CUDA 11, and on TensorFlow 2.3.1 or higher")
+    validate_or_skip_smdataparallel(n_virginia_ecr_image)
+    validate_or_skip_smdataparallel_efa(n_virginia_ecr_image)
```
Comment (saimidu) [inline test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smdataparallel.py:97]: So the multi-node smdataparallel test would no longer be usable for TF 2.3 images. It will only be applicable on tests which have EFA support, i.e., TF 2.4 and above.

Is this intended?

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smdataparallel.py [hunk 16]
```diff
@@ -100,10 +121,8 @@ def test_smdataparallel_throughput(instance_types, n_virginia_ecr_image, py_vers
     """
     Tests smddprun throughput
     """
-    _, image_framework_version = get_framework_and_version_from_tag(n_virginia_ecr_image)
-    image_cuda_version = get_cuda_version_from_tag(n_virginia_ecr_image)
-    if Version(image_framework_version) < Version("2.4.1") or image_cuda_version != "cu110":
-        pytest.skip("EFA is only supported on CUDA 11, and on TensorFlow 2.4.1 or higher")
+    validate_or_skip_smdataparallel(n_virginia_ecr_image)
```
Comment (saimidu) [inline test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smdataparallel.py:124]: Why run the less restrictive validation at all when the more restrictive validation for `efa` filters out more cases?

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smmodelparallel.py [hunk 17]
```diff
@@ -35,10 +60,8 @@ def test_smmodelparallel_efa(n_virginia_sagemaker_session, efa_instance_type, n_
     """
     Tests SM Modelparallel in sagemaker
     """
-    _, image_framework_version = get_framework_and_version_from_tag(n_virginia_ecr_image)
-    image_cuda_version = get_cuda_version_from_tag(n_virginia_ecr_image)
-    if Version(image_framework_version) < Version("2.4.1") or image_cuda_version != "cu110":
-        pytest.skip("Model Parallelism with EFA only supports CUDA 11, and on TensorFlow 2.4.1 or higher")
+    validate_or_skip_smmodelparallel(n_virginia_ecr_image)
```
Comment (saimidu) [inline test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smmodelparallel.py:63]: Why run the less restrictive validation at all when the more restrictive validation for `efa` filters out more cases?

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py [hunk 18]
```diff
@@ -237,6 +243,7 @@ def test_sanity_efa(n_virginia_ecr_image, efa_instance_type, n_virginia_sagemake
     Tests pt mnist command via script mode
     """
     validate_or_skip_smmodelparallel(n_virginia_ecr_image)
+    validate_or_skip_smmodelparallel_efa(n_virginia_ecr_image)
```
Comment (saimidu) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py:246]: Why run the less restrictive validation at all when the more restrictive validation for `efa` filters out more cases?

**Review by saimidu — APPROVED** [review]

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smdataparallel.py (hunk 15 above)
Comment (jeet4320) [reply test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smdataparallel.py:97]: it is not multinode, it just using two instances one by one. 
There is same non efa test as below which will run for tf2.3 (also it runs for above tf2.3)

https://github.com/aws/deep-learning-containers/pull/1044/files#diff-dd278248d0eb52756b3bf592fca639d25738021b922d8f4e4d4f5d54a52ba216R60

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smdataparallel.py (hunk 16 above)
Comment (jeet4320) [reply test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smdataparallel.py:124]: you are right, will remove validate_or_skip_smdataparallel/validate_or_skip_smmodelparallel for all efa tests

**Review by saimidu — APPROVED** [review]
